pkg/ottl/ottlfuncs/func_decode.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/decode"
```

Comment:
If possible I'd like to avoid pulling in a dependency on Stanza. Could we move common code to `coreinternal` in a separate PR? I think what we move depends on what we'd like to use from this package. We could either move just the encoding lookup, or the whole decoder.

Reply:
removing the dependency to stanza again would be fine with me (this will also significantly decrease the size of that PR again as including this new dependency required an update of all modules using the ottl package). If we agree to remove this dependency again, I would suggest to move the shared logic in a follow up PR - would that work for you? CC @TylerHelmuth

Reply:
We could at least copy the encoding overrides (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/decode/decode.go#L49) to `coreinternal`. Both Stanza and OTTL already depend on it.

Changing the encoding mapping later would be a breaking change, so unfortunately we will at least need to handle that here.

Reply:
I agree that OTTL shouldn't depend on stanza directly.